contracts/UniversalRouter.sol

```diff
@@ -79,6 +60,8 @@ contract UniversalRouter is IUniversalRouter, Dispatcher, RewardsCollector {
         return command & Commands.FLAG_ALLOW_REVERT == 0;
     }
 
-    /// @notice To receive ETH from WETH and NFT protocols
-    receive() external payable {}
+    /// @notice To receive ETH from WETH
+    receive() external payable {
+        if (msg.sender != address(WETH9)) revert InvalidEthSender();
```

Comment:
we'll eventually need to add V4 in here obvi, but yay for finally getting this in!

Reply:
yeah when i add v4 to the code i'll look at how best to do this